src/typing/generic.ml

```diff
@@ -145,6 +145,31 @@ let get_short_name =
 		Printf.sprintf "Hx___short___hx_type_%i" !i
 	)
 
+let static_method_container gctx c cf p =
+	let ctx = gctx.ctx in
+	let pack = fst c.cl_path in
+	let name = (snd c.cl_path) ^ "_" ^ cf.cf_name ^ "_" ^ gctx.name in
+	try
+		let t = Typeload.load_instance ctx ({ tpackage = pack; tname = name; tparams = []; tsub = None },p) true in
+		match t with
+		| TInst(cg,_) -> cg
+		| _ -> error ("Cannot specialize @:generic static method because the generated type name is already used: " ^ name) p
```

Comment:
Why do we care about the exact name here? Can't we just somehow enumerate the names?

Reply:
Because the name is used to request already generated type on sequential calls to the generic method.
It's the same mechanism used for current generic methods and for generic types:
https://github.com/HaxeFoundation/haxe/blob/40a3a762c0fe90fa768627d7c7227196463e13f5/src/typing/generic.ml#L171-L174
Otherwise we will have to store a list of generated types somewhere.